### PR TITLE
Vast SET improvements & new property operators

### PR DESCRIPTION
**Pull request by petraselmer**
This has been reviewed offline by Nigel. Please merge and fwd-merge.

### CODE REVIEW

**Comment by petraselmer** [comment]
This has been reviewed and approved by @nigelsmall

**Comment by petraselmer** [comment]
@jjaderberg still need to expunge PID data from here, hence putting DO NOT MERGE. Will sort out in 15 - 20 mins

**Comment by petraselmer** [comment]
@jjaderberg PID removed... so ready for merging

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/RemoveTest.scala
Comment (jjaderberg) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/RemoveTest.scala]: "from a node or relatinoship" or "from a graph element". I prefere the former, I'm not used to "graph element". In fact I'd prefer "from a node or relationship." on line 69 as well.

File: cypher/cypher-docs/src/docs/dev/syntax/operators.asciidoc
Comment (jjaderberg) [inline cypher/cypher-docs/src/docs/dev/syntax/operators.asciidoc]: If this was removed in 2.0, how much longer should we carry this note on in the documentation?

Comment (jjaderberg) [reply cypher/cypher-docs/src/docs/dev/syntax/operators.asciidoc]: Does Cypher give a warning on this usage? If so I think we should just remove this. If not, well maybe it could? Either case I think this should go soon.

**Review by jjaderberg — CHANGES_REQUESTED** [review]
Let's talk about "graph elements" before merging this

**Review by jjaderberg — APPROVED** [review]

**Pull request merged**
